File: mobile/lib/models/game/chapter.dart
class Chapter {
  const Chapter({
    required this.chapterId,
    required this.title,
    required this.description,
    required this.difficulty,
    required this.scenarioCount,
    required this.orderIndex,
  });

  factory Chapter.fromJson(Map<String, dynamic> json) {
    return Chapter(
      chapterId: json['chapterId'] as int,
      title: json['title'] as String,
      description: json['description'] as String,
      difficulty: json['difficulty'] as int,
      scenarioCount: json['scenarioCount'] as int,
      orderIndex: json['orderIndex'] as int,
    );
  }

  final int chapterId;
  final String title;
  final String description;
  final int difficulty;
  final int scenarioCount;
  final int orderIndex;
}

File: backend/src/main/java/com/phishingdefense/backend/entity/Chapter.java
package com.phishingdefense.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "chapters")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Chapter {

    @Id
    @Column(name = "chapter_id")
    private Integer chapterId;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "difficulty")
    private Integer difficulty;

    @Column(name = "scenario_count")
    private Integer scenarioCount;

    @Column(name = "order_index")
    private Integer orderIndex;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}
